Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -66,41 +109,59 @@ final class TransactionPoster: TransactionPosterType {
         productsManager: ProductsManagerType,
         receiptFetcher: ReceiptFetcher,
         backend: Backend,
+        cache: PostedTransactionCacheType,
         paymentQueueWrapper: EitherPaymentQueueWrapper,
         systemInfo: SystemInfo,
         operationDispatcher: OperationDispatcher
     ) {
         self.productsManager = productsManager
         self.receiptFetcher = receiptFetcher
         self.backend = backend
+        self.cache = cache
         self.paymentQueueWrapper = paymentQueueWrapper
         self.systemInfo = systemInfo
         self.operationDispatcher = operationDispatcher
     }
 
+    // TODO: consumables
```

Comment:
Gotta consider how this works with #2841.

Reply:
maybe we should get both into an integration branch before shipping and test them together?